src/core/Akka.API.Tests/CoreAPISpec.ApproveClusterSharding.approved.txt

```diff
@@ -156,70 +179,42 @@ namespace Akka.Cluster.Sharding
+    public sealed class ShardRegion : Akka.Actor.ActorBase, Akka.Actor.IWithTimers
     {
-        public readonly string ShardId;
-        public ShardInitialized(string shardId) { }
-        public bool Equals(Akka.Cluster.Sharding.ShardInitialized other) { }
-        public override bool Equals(object obj) { }
-        public override int GetHashCode() { }
-        public override string ToString() { }
-    }
-    public class ShardRegion : Akka.Actor.ActorBase, Akka.Actor.IWithTimers
-    {
-        public readonly Akka.Cluster.Cluster Cluster;
-        public readonly string CoordinatorPath;
-        public readonly System.Func<string, Akka.Actor.Props> EntityProps;
-        public readonly Akka.Cluster.Sharding.ExtractEntityId ExtractEntityId;
-        public readonly Akka.Cluster.Sharding.ExtractShardId ExtractShardId;
-        public readonly object HandOffStopMessage;
-        protected System.Collections.Immutable.IImmutableSet<Akka.Actor.IActorRef> HandingOff;
-        protected System.Collections.Immutable.IImmutableSet<Akka.Cluster.Member> MembersByAge;
-        protected System.Collections.Immutable.IImmutableDictionary<string, Akka.Actor.IActorRef> RegionByShard;
-        protected System.Collections.Immutable.IImmutableDictionary<Akka.Actor.IActorRef, System.Collections.Immutable.IImmutableSet<string>> Regions;
-        public readonly Akka.Cluster.Sharding.ClusterShardingSettings Settings;
-        protected System.Collections.Immutable.IImmutableDictionary<string, System.Collections.Immutable.IImmutableList<System.Collections.Generic.KeyValuePair<object, Akka.Actor.IActorRef>>> ShardBuffers;
-        protected System.Collections.Immutable.IImmutableDictionary<string, Akka.Actor.IActorRef> Shards;
-        protected System.Collections.Immutable.IImmutableDictionary<Akka.Actor.IActorRef, string> ShardsByRef;
-        protected System.Collections.Immutable.IImmutableSet<string> StartingShards;
-        public readonly string TypeName;
-        public ShardRegion(string typeName, System.Func<string, Akka.Actor.Props> entityProps, Akka.Cluster.Sharding.ClusterShardingSettings settings, string coordinatorPath, Akka.Cluster.Sharding.ExtractEntityId extractEntityId, Akka.Cluster.Sharding.ExtractShardId extractShardId, object handOffStopMessage, Akka.Actor.IActorRef replicator, int majorityMinCap) { }
-        public bool GracefulShutdownInProgress { get; }
-        public Akka.Event.ILoggingAdapter Log { get; }
-        protected object RegistrationMessage { get; }
+        public ShardRegion(string typeName, System.Func<string, Akka.Actor.Props> entityProps, Akka.Cluster.Sharding.ClusterShardingSettings settings, string coordinatorPath, Akka.Cluster.Sharding.ExtractEntityId extractEntityId, Akka.Cluster.Sharding.ExtractShardId extractShardId, object handOffStopMessage, Akka.Cluster.Sharding.Internal.IRememberEntitiesProvider rememberEntitiesProvider) { }
```

Comment:
Avoid changing an existing public constructor signature, this should be made into a constructor overload.

Reply:
not possible to keep existing signature here. `ShardRegion` should be internal anyway. Only reason it's not is because there are some nested messages inside, which need to be public